### PR TITLE
Replace boost::timer

### PR DESCRIPTION
**Pull request by geoffthemedio**
As reported [here](https://www.freeorion.org/forum/viewtopic.php?f=9&t=11501) `boost::timer` is deprecated. This adapts `ScopedTimer` to replace it.

### CODE REVIEW

File: UI/TechTreeWnd.cpp [hunk 1]
```diff
@@ -1868,9 +1869,9 @@ void TechTreeWnd::TechListBox::Update(bool populate /* = true */) {
         auto& tech_row = row.second;
         if (TechVisible(tech_row->GetTech(), m_categories_shown, m_tech_statuses_shown)) {
             tech_row->Update();
-            insertion_timer.restart();
+            insertion_timer.start();
             auto listbox_row_it = Insert(tech_row);
-            insertion_elapsed += insertion_timer.elapsed();
+            insertion_elapsed += ElapsedSeconds(insertion_timer);
```
Comment (geoffthemedio) [inline UI/TechTreeWnd.cpp]: probably need an optional parameter to control rounding to avoid issues when accumulating the results like this...

**Comment by geoffthemedio** [comment]
@adrianbroher @o01eg Some help with CMake would be appreciated. Apparently adding boost timer to the linked and required libs fixes the XCode build but breaks GCC? (Assuming the Travis CI build for XCode depends on CMake defs, which perhaps it doesn't, in which case I'm even more confused...)

**Comment by adrianbroher** [comment]
> @adrianbroher @o01eg Some help with CMake would be appreciated. Apparently adding boost timer to the linked and required libs fixes the XCode build but breaks GCC? (Assuming the Travis CI build for XCode depends on CMake defs, which perhaps it doesn't, in which case I'm even more confused...)

Seems like the boost timer library is not installed within the Docker image of the SDK.

**Comment by o01eg** [comment]
@geoffthemedio @adrianbroher I suppose we should test boost version before add timer component. Also, should we add this PR to release?

**Comment by adrianbroher** [comment]
Why? The CI setup has the package not installed, it's not like it is available in older boost releases.

**Comment by adrianbroher** [comment]
@geoffthemedio Why don't you use the Scoped{,Section}Timer?  I don't see any place in this change where it could not be used.

**Comment by geoffthemedio** [comment]
> @geoffthemedio Why don't you use the Scoped{,Section}Timer?

Done. Needed to add to the interface of `ScopedTimer` to be equivalent to `boost::timer`.

Also adapted `SectionedScopedTimer` to replace to re-use of `ScopedTimer`.

Also tweaked output formatting and options.

**Pull request merged**